ros2component/ros2component/api/__init__.py

```diff
@@ -182,13 +184,7 @@ def unload_component_from_container(*, node, remote_container_node_name, compone
             future = unload_node_client.call_async(request)
             rclpy.spin_until_future_complete(node, future)
             response = future.result()
-            if not response.success:
-                return 'Failed to unload component {} from {} container\n  {}'.format(
-                    uid, remote_container_node_name, response.error_message
-                )
-            print('Unloaded component {} from {} container'.format(
-                uid, remote_container_node_name
-            ))
+            yield uid, not response.success, response.error_message
```

Comment:
Why `yield` here? I don't see any `async` anywhere around for this function.

Reply:
Because it's a generator.

Reply:
Oh, I see it in the `for` loop now. :+1: